app/uk/gov/hmrc/customs/declarations/information/connectors/declarationsInformationConnectors.scala

```diff
@@ -127,14 +125,37 @@ abstract class DeclarationConnector @Inject()(http: HttpClient,
               correlationId: CorrelationId,
               apiVersion: ApiVersion,
               maybeApiSubscriptionFieldsResponse: Option[ApiSubscriptionFieldsResponse],
-              searchType: SearchType)(implicit asr: AuthorisedRequest[A]): Future[HttpResponse] = {
+              searchType: SearchType)(implicit asr: AuthorisedRequest[A]): Future[Either[DeclarationConnector.Error, HttpResponse]] = {
 
     val config = Option(serviceConfigProvider.getConfig(s"${apiVersion.configPrefix}$configKey")).getOrElse(throw new IllegalArgumentException("config not found"))
+    val url = config.url
     val bearerToken = "Bearer " + config.bearerToken.getOrElse(throw new IllegalStateException("no bearer token was found in config"))
     val headers: Seq[(String, String)] = getHeaders(date, asr.conversationId, correlationId) ++ Seq((AUTHORIZATION, bearerToken))
 
     val declarationPayload = backendPayloadCreator.create(asr.conversationId, correlationId, date, searchType, maybeApiSubscriptionFieldsResponse)
-    withCircuitBreaker(post(declarationPayload, config.url, headers))
+
+    case class Non2xxResponseException(status: Int, responseBody: String) extends Throwable
+    withCircuitBreaker {
+      logger.debug(s"Sending request to $url. Headers $headers Payload: ${declarationPayload.toString()}")
```

Comment:
Could [] be put around the log variables?

Reply:
Yup, fixed.